cgaal-engine/src/algorithms/game_strategy/partial.rs

```diff
@@ -31,241 +32,265 @@ impl PartialStrategy {
     }
 }
 
-/// Computes the PartialStrategy that the given players can use to enforce or break
-/// the given property. In other words, the behaviour of this function is undefined, if the
-/// ATL formula of v0 is an unsatisfied enforce property or a satisfied despite property.
-pub fn compute_partial_strategy<G: GameStructure>(
+/// Compute the moves of the witness strategy.
+/// This function handles the case where the formula is an enforce formula assigned true.
+/// Result is not minimal.
+pub fn find_strategy_moves_true_case<G: GameStructure>(
     graph: &AtlDependencyGraph<G>,
     v0: &AtlVertex,
     assignments: &HashMap<AtlVertex, VertexAssignment>,
-) -> PartialStrategy {
-    let mut move_to_pick = HashMap::new();
+) -> HashMap<State, PartialMove> {
+    debug_assert!(v0.formula().is_enforce());
+    debug_assert_eq!(assignments[v0], VertexAssignment::True);
 
-    compute_partial_strategy_rec(graph, v0, assignments, &mut move_to_pick);
+    if v0.formula().is_invariant() {
+        // EnforceInvariant has negation edge to DespiteEventually assigned undecided
+        let edges = graph.succ(v0);
+        let target = &edges[0].targets()[0];
+        return find_strategy_moves_undecided_case(graph, target, assignments);
+    }
 
-    PartialStrategy {
-        players: v0.formula().players().unwrap().into(),
-        move_to_pick,
+    if v0.formula().is_next() {
+        // Find first edge with all targets true
+        for edge in graph.annotated_succ(v0) {
+            let all_targets_true = edge
+                .targets()
+                .iter()
+                .all(|(target, _)| assignments[target].is_true());
+            if all_targets_true {
+                let mut move_to_pick = HashMap::<State, PartialMove>::new();
```

Comment:
This variable name seems to only refer to a single move - but as I now understand is that is actually a map of states and partial moves. I suggest we rename it.

Reply:
I am not sure what you are arguing. Do you want to change it because it only has one entry? Or do you want to change it because it can hold more entries?

All the strategy synthesis functions in `partial.rs` return a map describing which partial move the coalition must pick in each state. In all the functions, the computed map is consistently called `move_to_pick` no matter the number of entries, since it describes which "move to pick" for each state.

Reply:
It's nitpicky and could be either (actually a third)

> In all the functions, the computed map is consistently called move_to_pick no matter the number of entries, since it describes which "move to pick" for each state.

This gives rise to the naming `move_to_pick_for_state` or something equally verbose.

Reply:
I understand that its consistently named `move_to_pick` but I suggest we find another name since it is ambigious in the way it can refer to a "single" move, and that is not the case, but instead it refer to a map. 

IF we choose to rename it ofcourse it should be renamed consistently.

Reply:
"move" is singular because it assignments a single move to each state